### PR TITLE
Bump to sqlite version 3.31.0

### PR DESCRIPTION
**Pull request by M66B**
Build/deploy was not tested, but all links where checked.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ Use new SQLite features:
 - **[JSON1 extension](https://www.sqlite.org/json1.html)**
 - **[Common Table expressions](https://www.sqlite.org/lang_with.html)**
 - **[Indexes on expressions](https://www.sqlite.org/expridx.html)**
+- **[Full Text Search 5](https://www.sqlite.org/fts5.html)**
```
Comment (mikehardy) [inline README.md]: I don't believe this is accurate unless some further action is taken in the build system here?

https://www.sqlite.org/fts5.html

> 2.1. Building FTS5 as part of SQLite
As of version 3.9.0 (2015-10-14), FTS5 is included as part of the SQLite amalgamation. It is disabled by default. If using the two autoconf build system, it is enabled by specifying the "--enable-fts5" option when running the configure script.

Or, if sqlite3.c is compiled using some other build system, by arranging for the SQLITE_ENABLE_FTS5 pre-processor symbol to be defined.

**Comment by mikehardy** [comment]
I hate just pooping on a PR without saying thanks to anyone proposing any PR in an open source project, so thank you :-)

**Comment by M66B** [comment]
I am using FTS5 successfully in [FairEmail](https://github.com/M66B/FairEmail/), so I don't think changes are required.

Source code: https://github.com/M66B/FairEmail/blob/master/app/src/main/java/eu/faircode/email/FtsDbHelper.java

**Comment by mikehardy** [comment]
Fascinating! That's unexpected based on the docs :thinking:

**Comment by M66B** [comment]
It seems to me that building FTS5 is being enabled here:

https://github.com/requery/sqlite-android/blob/master/sqlite-android/src/main/jni/sqlite/Android.mk#L23

**Comment by mikehardy** [comment]
and that explains it! good find - consistent docs, LGTM then for whatever that's worth :-)

**Comment by M66B** [comment]
Maybe it is worth to mention [Generated Columns](https://www.sqlite.org/gencol.html) too (a brand new feature).

**Comment by mikehardy** [comment]
Up to you, I just fuzzbust PRs here in an attempt to help @npurushe :-)

**Comment by M66B** [comment]
I have added generated columns to the description and update the native library sizes as well.

**Comment by npurushe** [comment]
Thanks will merge and deploy this change soon

**Pull request merged**